metricbeat/module/kubernetes/pod/data.go

```diff
@@ -167,6 +150,7 @@ func eventMapping(content []byte, metricsRepo *util.MetricsRepo, logger *logp.Lo
 		}
 
 		if usageMem > 0 {
```

Comment:
Here we dont check if podMemLimit > 0, so this means that memory.usage.node.pct will always be reported for a pod right? 

But what we need is to report it only in case that `podMemLimit == 0`

Otherwise this [Kibana UI](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/metrics_data_access/common/inventory_models/kubernetes/pod/metrics/tsvb/pod_memory_usage.ts#L27) will always use the node limits

Reply:
> Here we dont check if podMemLimit > 0, so this means that memory.usage.node.pct will always be reported for a pod right?

no, `memory.usage.node.pct` will be reported only in case `nodeMem > 0` (`nodeMem` can be 0 if `state_node` and/or `node` metricsets are disabled)

> But what we need is to report it only in case that `podMemLimit == 0`

why? the point is to split `memory.usage.node.pct` and `memory.usage.limit.pct`, so if `usageMem` and `nodeMem` are available - `memory.usage.node.pct` can be calculated
similarly: if all containers have memory limit defined, `memory.usage.limit.pct` can be calculated as usageMem/sum(containers_mem_limit)

> Otherwise this [Kibana UI](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/metrics_data_access/common/inventory_models/kubernetes/pod/metrics/tsvb/pod_memory_usage.ts#L27) will always use the node limits

I think Kibana issue should be covered separately by https://github.com/elastic/kibana/issues/184984

Reply:
Ok, I had a second look [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/metrics_data_access/common/inventory_models/kubernetes/pod/metrics/tsvb/pod_memory_usage.ts#L43). So this change wont break the existing UI, because kibana checks kubernetes.pod.memory.usage.limit.pct > 0.0. As long as this metric is populated then we dont care for the node.pct.

Reply:
hm, it seems to not work this way
<img width="1195" alt="Screenshot 2024-07-02 at 00 09 54" src="https://github.com/elastic/beats/assets/28299531/0a9634b1-86d8-4023-950d-4a1235697a1b">

usage.node.pct is > 1% for the selected pod:
<img width="1511" alt="Screenshot 2024-07-02 at 00 11 07" src="https://github.com/elastic/beats/assets/28299531/2e8de451-403b-4c0f-a722-192b958e25fe">

seems this [script](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/metrics_data_access/common/inventory_models/kubernetes/pod/metrics/tsvb/pod_memory_usage.ts#L43) doesn't work as expected